mlflow/tracking/fluent.py

```diff
@@ -582,8 +583,20 @@ def log_param(key: str, value: Any) -> Any:
     :param value: Parameter value (string, but will be string-ified if not).
                   All backend stores support values up to length 500, but some
                   may support larger values.
-
-    :return: the parameter value that is logged.
+    :param synchronous: [Experimental] Defaults to True.
+                        Indicates if the parameter would be logged in synchronous
+                            fashion or not.
+                        When it is True this call would be blocking call and offers immediate
+                            consistency of the parameter upon returning.
+                        When this value is set to False, parameter would be logged in async
+                            fashion. Backing provider gurantees that metrics are accepted
+                            into system but would persist them with some time delay -
+                            'near-real time'/'eventual' consistency.
+
+    :return: When synchronous=True, the parameter value that is logged.
+                When synchronous=False, returns
+                :py:class:`mlflow.utils.async_logging.run_operations.RunOperations` object.
```

Comment:
@chenmoneygithub @sagarsumant can we make sure that `RunOperations` appears in our documentation? Else, this is a broken link and users can't see what actions can be performed on a `RunOperations` object.

I think we can import `RunOperations` in the top-level `mlflow` module and document it there.

Reply:
I have made changes as per my understanding, some may not be reflected as you would expect, but will check with @chenmoneygithub to get it fixed, if something is still missing.